src/goto-cc/gcc_mode.cpp

```diff
@@ -813,7 +813,7 @@ int gcc_modet::run_gcc(const compilet &compiler)
   for(const auto &a : cmdline.parsed_argv)
     new_argv.push_back(a.arg);
 
-  if(compiler.wrote_object_files())
+  if(!act_as_ld && compiler.wrote_object_files())
```

Comment:
It is not clear to me where this is set / why this would work but in this part of the code I am willing to take your word for it.

Reply:
`act_as_ld` is a class member set upon construction, with its value dependent on the name of the executable that is running this code ("goto-ld" -> `true`, "goto-cc" -> `false`). When acting as a linker, we shouldn't be adding preprocessor command-line options.

Reply:
OK; fair.